src/lib/sidebar/scheme_data.ts

```diff
@@ -62,25 +61,52 @@ export function backfill(json: SchemeCollection) {
     }
   }
 
-  // Ensure every scheme has some color
+  let schema = get(schemaStore);
   for (let scheme of Object.values(json.schemes)) {
-    scheme.color ||= randomSchemeColor();
+    // Ensure every scheme has some color
+    (scheme as any).color ??= randomSchemeColor();
+
+    // Ensure pipeline defaults are set
+    if (schema == "pipeline") {
+      (scheme as any).pipeline ??= emptyPipelineScheme();
+      // Any changes to PipelineScheme after 21 November 2023 must be handled here
```

Comment:
There'll be a few in the next PR, but I expect it to be nice and simple to handle here only.